### ISSUE TITLE
NVME Boot UP

### ISSUE THREAD

**Issue by hutger** [hashicorp/terraform-provider-vsphere#1276]
### Description

Hi there,

I've created a template using Packer with three attached NVME disks. When I try to run Terraform for deploying an instance using this template, I get the message:

Error: there must be at least one disk specified

It seems that Terraform is not able to find the NVME disks attached (since the only opinions are pvscsi, buslogic, etc, but none NVME), even though I add an extra_config informing NVME is presenting:

disk_size_gb                    = var.vmRootDiskSize
scsi_type                       = var.vmScsiType
extra_config                    = {
    "nvme0.present"                 = "True"
    "nvme0.0.present"               = "True"

PS: If we deploy a new instance using the Template manually, the instance boots up as expected.

### References

As per the documentation, scsi_type NVME is not supported.
https://registry.terraform.io/providers/hashicorp/vsphere/latest/docs/resources/virtual_machine#scsi_type

### Community Note

Is the support for NVME on the roadmap (or maybe there's a workaround for now)?

Thanks for any suggestion or advice,

Hutger

**Comment by apollo13** [comment]
I do not think it makes sense to extend `scsi_type` to include NVME. it will need it's own `nvme_` options like `ide_` & `sata_`.

**Comment by hutger** [comment]
Hi @apollo13 , would you propose an alternative approach for rolling out instances with NVME attached?

**Comment by apollo13** [comment]
I don't think it is currently possibly with the terraform provider. If you do `terraform import` of an existing machine with nvme disks you will see that those are not discovered at all. I fear there are plenty of changes needed :/

**Comment by eizedev** [comment]
+1 for adding support for NVME controller

**Comment by rhoughton-pivot** [comment]
I am needing to recreate my template (made in Packer, also a HashiCorp project) to not use nvme because of this issue

**Comment by alexanderek** [comment]
When will this be supported?

**Comment by vash1028** [comment]
Any update on this?

**Comment by lostplan** [comment]
+1 this would be really useful

**Comment by johnypony3** [comment]
hi @tenthirtyam, any chance this can be re-prioritized? nvme is available to be built via packer but not deployed via terraform. im being asked to implement this. this ticket took over a year to get acknowledged.

thanks!

**Comment by rjungnitz** [comment]
Hi @tenthirtyam, I'd like to 2nd @johnypony3 request.  Our understanding of the nvme controller is that it improves performance when the backend disk is NVMe SSD.  We are trying to build high performance Oracle and MS SQL database systems.

**Comment by tenthirtyam** [comment]
This can be researched for an upcoming milestone based on the number of community reactions 👍.

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

**Comment by patrikcze** [comment]
+1 for adding support for NVME controller

**Comment by thewayofvikings** [comment]
+1 for adding support for NVME controller :thumbsup:

**Comment by tenthirtyam** [comment]
@patrikcze  and @thewayofvikings - upvotes are **only** counted on the opening description, _not comments_.


**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

**Comment by Hogusss** [comment]
Hi, is there a possibility to plan this feature for 2.4 ? It's a big show stopper for us to use Terraform in prod because nvme is our standard. Would be amazing to have it. 
thanks

**Comment by tigattack** [comment]
> Acknowledged. This can be researched for an upcoming milestone based on the number of community reactions 👍.
> 
> If you have not added your own 👍, please do so on the originating description.
> 
> **Ryan Johnson** Staff II Solutions Architect | VMware, Inc.

This seems to be the [most upvoted open issue in the repository](https://github.com/hashicorp/terraform-provider-vsphere/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-desc) at the moment, it would be great if it could be reconsidered for inclusion in a release milestone.

**Comment by rjungnitz** [comment]
I'm still in favor of adding support for the NVMe controller, but I just wanted to point out that depending on what backend storage is being used, the use of this controller can actually lead to translation back to SCSI, thus negatively impacting performance.  So check with your storage vendor before selecting this controller.

**Comment by JM2K69** [comment]
Any Update ??

**Comment by tenthirtyam** [comment]
> Any Update ??

Please refer to the attached milestone for this enhancement request.

**Comment by nathan-owen** [comment]
Yeah I get that contributions are welcome, but based on the backlog of pull requests, it doesn't seem like there's a lot of movement on this provider besides small bug fixes.

**Comment by tigattack** [comment]
After a year, this is still the [most upvoted open issue in the repository](https://github.com/hashicorp/terraform-provider-vsphere/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc), and by a decent margin too. I must wonder why it's still stuck in the backlog after so long if issues are in fact prioritised by the number of 👍 reacts, as the issue template claims.

Sure, it would be nice if someone from the community would contribute this feature, but I'd still hope that a first-party, "official" provider would be better supported by HashiCorp, especially given that the README explicitly states that such providers are maintained by HashiCorp's Terraform team.

**Comment by msproul-onezero** [comment]
+1 for adding support for NVME controller

**Comment by mrfootballnz** [comment]
+1 for adding support for NVME controller
